src/iuse.cpp

```diff
@@ -9187,6 +9187,79 @@ int iuse::disassemble( player *p, item *it, bool, const tripoint & )
     return 0;
 }
 
+int iuse::gobag( player *p, item *it, bool, const tripoint & )
+{
+    std::map<std::string, int> gobag_contents = {
+        { "helmet_army", 1 },
+        { "jacket_army", 1 },
+        { "pants_army", 1 },
+        { "tshirt", 3 },
+        { "under_armor", 1 },
+        { "chestrig", 1 },
+        { "coat_rain", 1 },
+        { "hood_rain", 1 },
+        { "long_underpants", 1 },
+        { "towel", 1 },
+        { "boots_combat", 1 },
+        { "socks", 3 },
+        { "gloves_tactical", 1 },
+        { "light_plus_battery_cell", 1 },
+        { "flashlight", 1 },
+        { "radio", 1 },
+        { "lighter", 1 },
+        { "pockknife", 1 },
+        { "multitool", 1 },
+        { "1st_aid", 1 },
+        { "rollmat", 1 },
+        { "sleeping_bag_roll", 1 },
+        { "ear_plugs", 1 },
+        { "mre_hashbrownbacon_box", 1 },
+        { "mre_macaronimarinara_box", 1 },
+        { "mre_meatball_box", 1 },
+        { "bottle_plastic", 2 }
+    };
```

Comment:
I'm not convinced there's a good reason to move this out of json.  Altering the contents now requires code changes/recompile and this is exactly the sort thing we have json parsing for.

Reply:
I'm not opposed to this but I'm not sure how to leave it in json and fix the bugs at the same time. Aside from #22622 and #24060, the lighter spawns with random charges and the batteries spawn with 0 charges. Is there a way to keep the item list with the item and fix all the bugs?

Reply:
Create an item group with specific amount of charges. If it is not allowed yet - change the code to make it possible.